2_process/src/munge_inst_timeseries.R

```diff
@@ -55,3 +55,36 @@ aggregate_data_to_hourly <- function(inst_data,output_tz){
 
 }
 
+aggregate_data_to_daily <- function(inst_data, daily_data, output_tz){
+  #' 
+  #' @description Function to aggregate instantaneous NWIS data collected at sub-hourly (e.g. 15/5/30 min) intervals to hourly min/mean/maxs
+  #'
+  #' @param inst_data a data frame containing the downloaded time series for NWIS instantaneous site. 
+  #' inst_data must include the following columns: c("Value_Inst","dateTime","site_no")
+  #' @param daily_data a data frame the downloaded daily time series of DO data. This is used so that if a site is already in 
+  #'  the daily sites, we won't do the aggregating here
+  #' @param output_tz character string to set display attribute of dateTime. Possible values to provide are "UTC",
+  #' "America/New_York","America/Chicago", "America/Denver","America/Los_Angeles", "America/Anchorage", 
+  #' as well as the following which do not use daylight savings time: "America/Honolulu", "America/Jamaica",
+  #' "America/Managua","America/Phoenix", and "America/Metlakatla"
+  #'
+  #' @value A data frame containing daily min, mean, and max values the parameter of interest
+  
+  
+  only_inst_data = setdiff(inst_data$site_no, daily_data$site_no)
+  
+  daily_values <- inst_data %>%
+    filter(site_no %in% only_inst_data) %>%
+    mutate(dateTime = with_tz(dateTime,tzone=output_tz)) %>%
+    mutate(Date = date(dateTime)) %>%
+    group_by(Date, site_no) %>%
+    summarise(Value = mean(Value_Inst, na.rm=TRUE), 
+              Value_Min = min(Value_Inst, na.rm=TRUE), 
+              Value_Max = max(Value_Inst,na.rm=TRUE), 
+              na_count=sum(is.na(Value_Inst)), 
+              value_count=sum(!is.na(Value_Inst))) %>%
```

Comment:
Can add `.groups = "keep"` to the summarize call to get rid of the warning message

Reply:
Odd. I haven't seen any warnings. I do see this though when I build with `.groups = "keep"`:
```R
* start target p2_inst_data_daily
`summarise()` has grouped output by 'Date', 'site_no'. You can override using the `.groups` argument.
```
w/o that I get 
```R
* start target p2_inst_data_daily
`summarise()` has grouped output by 'Date'. You can override using the `.groups` argument.
```

Reply:
I guess it's not a warning, but just a message from {dplyr}. I'm surprised you see those messages when you build with .groups="keep". Out of curiosity do you have a comma in between the last command and .groups = "keep" (`value_count=sum(!is.na(Value_Inst)), .groups="keep"`)

Reply:
I have this:
```R
group_by(Date, site_no, .groups = "keep") %>%
```

Reply:
Oh, sorry I realize my original comment wasn't very clear. `.groups="keep"` is an argument within the `summarize` step.

Reply:
```
summarize(Value = mean(Value_Inst,na.rm=TRUE), 
              Value_Min = min(Value_Inst,na.rm=TRUE), 
              Value_Max = max(Value_Inst,na.rm=TRUE), 
              na_count=sum(is.na(Value_Inst)), 
              value_count=sum(!is.na(Value_Inst)),
              .groups="keep")
```

Reply:
No problem. Thanks for the correction. That just shows my newbie-ness in R :)

So I added that in there and now I don't get any message at all.

Do you mind briefly explaining why that is? What does `.groups = "keep"` actually even do?

Reply:
`.groups` controls the output structure of the data frame, so specifying `.groups="keep"` results in a "grouped tibble" of class `"grouped_df"` (i.e. the grouping is retained). I pretty much use it to silence the output message from dplyr. We could also specify `.group="drop"` to return an ungrouped tibble/data frame.

Reply:
Ah. That makes sense. Thanks for explaining.